DDFLSBP-651 - Implement cookie blocking placeholder

Dresse opened the pull request:
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-651

#### Description

This PR adds the Cookie placeholder element to the design system. 
Link to the design in Figma: https://www.figma.com/design/Zx9GrkFA3l4ISvyZD2q0Qi/Designsystem?node-id=16042-22933&t=kwXRJW0pZzoQmJHm-1

It is added as part of the VideoEmbed story.

#### Screenshot of the result
Image showing the result in dpl-desgin-system storybook:

Desktop:
![Screenshot 2024-06-06 at 16 19 24](https://github.com/danskernesdigitalebibliotek/dpl-design-system/assets/6142323/2320ddd2-1972-4ab3-aaba-70fca687df41)

Mobile:
![Screenshot 2024-06-06 at 16 18 53](https://github.com/danskernesdigitalebibliotek/dpl-design-system/assets/6142323/abf13ec8-c5fd-4565-834b-3c055e97be66)

Video recording showing the end result:
https://github.com/danskernesdigitalebibliotek/dpl-design-system/assets/6142323/74dd4689-ad6f-4684-80e1-05eed9fb2f8c

Adamik10 commented on src/stories/Library/cookie-placeholder/CookiePLaceholder.stories.tsx:
```diff
@@ -0,0 +1,31 @@
+  parameters: {
+    design: {
+      type: "figma",
+      url: "https://www.figma.com/design/Zx9GrkFA3l4ISvyZD2q0Qi/Designsystem?node-id=16042-22933&t=kwXRJW0pZzoQmJHm-1",
```
Great with a link to Figma 👏✨

Adamik10 commented on src/stories/Library/cookie-placeholder/CookiePlaceholder.tsx:
```diff
@@ -0,0 +1,31 @@
+        <div id="dpl-react-apps-cookie-placeholder">{info}</div>
+      </div>
+
+      <Button
```
Wonderful that you're reusing existing components ✌️

Adamik10 commented on src/stories/Library/cookie-placeholder/cookie-placeholder.scss:
```diff
@@ -0,0 +1,14 @@
+  border-radius: 5px;
+}
+
+.cookie-placeholder__description {
```
Great BEM structure! 💃🏼

Adamik10 commented on src/stories/Library/cookie-placeholder/cookie-placeholder.scss:
```diff
@@ -0,0 +1,14 @@
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  padding: 20px;
```
If we have a variable in the spacing-variables.scss file that is close to this, just use that instead, please. Otherwise it's okay. But we generally prefer using the variables for px values if there is anything close by (+-2px for small things / +- 10px for larger values)

Dresse replied:
I have replaced the padding: 20px and margin:bottom with a suitable variable from variables-spacing.scss.

Adamik10 reviewed (changes requested):
I have one comment I'd like you to look at :)) Otherwise great job!

Also for future reference, I would like to see a mobile view screenshot in the PR overview. 

✨🎖️

Dresse commented:
> I have one comment I'd like you to look at :)) Otherwise great job!
> 
> Also for future reference, I would like to see a mobile view screenshot in the PR overview.
> 
> ✨🎖️

Thanks for all the feedback and comments Adam 👍 

I have updated the PR with new screenshots (both desktop and mobile).

I have also changed the padding: 20px and margin-bottom: 20px with a suitable variable from variables-spacing.scss. 

Please take a another look 😄

spaceo reviewed (changes requested):
I honour the review already given.
But I have one comment I think is, kind of, important.
I think the cookie placeholder story and the video embed story should me merged. Either could the cookie placeholder story be a sub story of the video embed or you could have a control on the videoembed story that toggels the placeholder.
There is alreday a whole lot of stories in the storybook and I think we can try to "minmize clutter" by bundling  stories that are related.

spaceo commented:
Btw there is a whole lot of chromatic tests that need to be approved. I think at least some of them are related. One troubeling one is one that seems like the contrast changes on the input fields that Claus did has been reverted?:
https://www.chromatic.com/test?appId=616ffdab9acbf5003ad5fd2b&id=6661c5837b4c653d0766aa3c

Dresse commented:
> I honour the review already given. :) But I have one comment I think is, somewhat, important: I think the cookie placeholder story and the video embed story should me merged. Either could the cookie placeholder story be a sub story of the video embed or you could have a control on the videoembed story that toggels the placeholder. There is alreday a whole lot of stories in the storybook and I think we can try to "minmize clutter" by bundling stories that are related.

I have now added the cookie placeholder story to the videoEmbed story. It has a control to toggle between the video and the placeholder. You can also click on the button on the placeholder to "accept" cookies and then see the video.

Dresse commented:
> Btw there is a whole lot of chromatic tests that need to be approved. I think at least some of them are related. One troubeling one is one that seems like the contrast changes on the input fields that Claus did has been reverted?: https://www.chromatic.com/test?appId=616ffdab9acbf5003ad5fd2b&id=6661c5837b4c653d0766aa3c

It seems to be because I didn't have the newest version of develop merged into my branch. After merging develop into my branch, the only chromatic tests that needs to be approved are the new one introduces by my PR.

Adamik10 reviewed (approved):
Wonderful 💯 

To answer your question about visibility in storybook - yes, I would probably give it an argument in storybook that overwrites some styling in order to show the component. But then we would also need to make sure people don't take it into dpl-react and add it there for correct rendering in the final product :)

Dresse commented:
> Wonderful 💯
> 
> To answer your question about visibility in storybook - yes, I would probably give it an argument in storybook that overwrites some styling in order to show the component. But then we would also need to make sure people don't take it into dpl-react and add it there for correct rendering in the final product :)

This is not relevant anymore as I have added it as part of the VideoEmbed story. I added functionality to hide/show the placeholder when clicking on the button and also as part of a config.
